server/lib/tuist/qa/agent.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Tuist.QA.Agent do
+  @moduledoc """
+  Tuist QA agent module.
+  """
+
+  alias Tuist.QA.SimulatorController
+  alias Tuist.Zip
+
+  def test(%{preview_url: preview_url, bundle_identifier: bundle_identifier}) do
+    run_preview(preview_url, bundle_identifier)
+  end
+
+  defp run_preview(preview_url, bundle_identifier) do
+    with {:ok, preview_path} <- download_preview(preview_url),
+         {:ok, device} <- simulator_device(),
+         {:ok, app_path} <- extract_app_from_preview(preview_path, bundle_identifier),
+         :ok <- SimulatorController.boot_simulator(device),
+         :ok <- SimulatorController.install_app(app_path, device) do
+      SimulatorController.launch_app(bundle_identifier, device)
+    end
+  end
+
+  defp simulator_device do
+    case SimulatorController.devices(
+           runtime_identifier: "com.apple.CoreSimulator.SimRuntime.iOS-18-5",
+           name: "iPhone 16"
+         ) do
+      {:ok, []} ->
+        {:error, "No simulator found with name 'iPhone 16' for runtime 'com.apple.CoreSimulator.SimRuntime.iOS-18-5'"}
+
+      {:ok, [device | _]} ->
+        {:ok, device}
+
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  defp download_preview(preview_url) do
+    with {:ok, preview_path} <- Briefly.create(extname: ".zip"),
```

Comment:
The lifecycle of these pats are bound to the process from where this action takes place. Is it fine considering the architecture that you have in mind?

Reply:
yup, that's fine